fix: rsvp-core TypeScript build errors (missing DOM types)

## Bug

The `packages/rsvp-core` build is failing with TypeScript errors:

```
error TS2304: Cannot find name 'window'.
error TS2304: Cannot find name 'clearTimeout'.
```

## Cause

The package is using browser globals (`window`, `setTimeout`, `clearTimeout`) but the tsconfig doesn't include DOM lib.

## Fix Options

1. Add `"lib": ["ES2020", "DOM"]` to tsconfig.json
2. Or use `typeof window !== 'undefined'` guards
3. Or use `globalThis` instead of `window`

## Priority

**BLOCKING** - This breaks all deploys.

## Files to check
- `packages/rsvp-core/tsconfig.json`
- `packages/rsvp-core/src/engine.ts` (lines 189, 211)

--- a/packages/rsvp-core/tsconfig.json
+++ b/packages/rsvp-core/tsconfig.json
@@ -1,6 +1,7 @@
 {
   "extends": "@tools/typescript-config/base.json",
   "compilerOptions": {
+    "lib": ["ES2020", "DOM"],
     "outDir": "./dist",
     "rootDir": "./src",
     "declaration": true,

